File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currMax = nums[0], subMax = nums[0];
        for(int i = 1; i < nums.length; i++) {
            currMax = Math.max(currMax + nums[i], nums[i]);
            subMax = Math.max(subMax, currMax);
        }
        return subMax;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        total_max, curr_max = float('-inf'), float('-inf')
        for num in nums:
            curr_max = max(curr_max + num, num)
            total_max = max(total_max, curr_max)
        return total_max